libs/platform/src/lib/components/menu/menu-item.component.ts

```diff
@@ -8,73 +8,105 @@ import {
     ChangeDetectionStrategy,
     OnDestroy,
     Input,
-    ViewEncapsulation
+    ViewEncapsulation,
+    ContentChild,
+    ChangeDetectorRef
 } from '@angular/core';
 import { FocusableOption } from '@angular/cdk/a11y';
 import { ENTER, SPACE } from '@angular/cdk/keycodes';
 import { Subject } from 'rxjs';
 
 import { KeyUtil } from '@fundamental-ngx/core/utils';
+import { MenuInteractiveDirective } from '@fundamental-ngx/core/menu';
 
 @Component({
     selector: 'fdp-menu-item',
     templateUrl: './menu-item.component.html',
     styleUrls: ['./menu-item.component.scss'],
     changeDetection: ChangeDetectionStrategy.OnPush,
-    encapsulation: ViewEncapsulation.None
+    encapsulation: ViewEncapsulation.None,
+    host: {
+        class: 'fd-menu__item',
+        tabindex: '-1'
+    }
 })
 export class MenuItemComponent implements OnDestroy, FocusableOption {
-    @Input() cascadeDirection: 'right' | 'left' = 'right';
+    @Input()
+    cascadeDirection: 'right' | 'left' = 'right';
 
     /** Event emitted when item is selected. */
-    @Output() itemSelect: EventEmitter<void> = new EventEmitter();
+    @Output()
+    itemSelect: EventEmitter<void> = new EventEmitter<void>();
 
     /** Track when menu item is hovered over */
-    public hovered: Subject<MenuItemComponent> = new Subject<MenuItemComponent>();
-
-    constructor(private elementRef: ElementRef) { }
-
-    /** @hidden set CSS class for host element. */
-    @HostBinding('class.fd-menu__item') menuItemClass = true;
-
-    /** @hidden set CSS class for host element. */
-    @HostBinding('class.fd-menu__link') menuLinkClass = true;
+    hovered: Subject<MenuItemComponent> = new Subject<MenuItemComponent>();
 
     /** Sets whether this item is a trigger for sub-menu. */
-    @HostBinding('class.trigger') isTrigger = false;
-
-    /** @hidden set CSS class 'is-selected' if menu-item opens a sub-menu. */
-    @HostBinding('class.is-selected')
-    isSelected = false;
+    @HostBinding('class.trigger')
+    isTrigger = false;
+
+    /** @hidden */
+    _isSelected = false;
+    /**
+     * @hidden
+     * set CSS class 'is-selected' if menu-item opens a sub-menu.
+     */
+    set isSelected(selected: boolean) {
+        const changed = selected !== this._isSelected;
```

Comment:
Why not use comparison directly in condition?

Reply:
moved it to the condition